compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1150,9 +1151,31 @@ impl<'tcx> TyCtxt<'tcx> {
 
         // This function modifies `self.definitions` using a side-effect.
         // We need to ensure that these side effects are re-run by the incr. comp. engine.
```

Comment:
Drive-by: could you update the comment before the call to `create_def`? Some parts are not accurate any more.